lib/Mojo/Message.pm

```diff
@@ -630,7 +630,10 @@ L</"default_charset">.
 
   my $str = $msg->to_string;
 
-Render whole message.
+Render whole message. For performance optimization reasons, calling this
+method freezes the message representation, so that subsequent calls
+to other methods (e.g. to modify the headers) will have no effect on the
+message rendering.
```

Comment:
This is incorrect, only certain parts of the message might get frozen, and we make no guarantees which. (It's an optimization after all, and we need to be able to keep changing it!)